pkg/dom0-ztools/rootfs/bin/eve

```diff
@@ -45,6 +73,36 @@ case "$1" in
   resume) [ -z "$2" ] && help
           ${CTR_CMD} t resume $2
           ;;
+ persist) case "$2" in
+               list) list_partitions
+                     ;;
+             attach) [ -z "$3" ] && help
+                     shift 2
+                     #fetch current P3 partition
+                     curr_p3_dev=$(/sbin/findfs PARTLABEL=P3)
+                     [ -z "$curr_p3_dev" ] && echo "Failed to find current P3 device" && exit 1
+                     curr_p3_dsk="/dev/"$(lsblk -no pkname "$curr_p3_dev")
+
+                     #rename current partition label
+                     relabel_dev "$p3_old_tag" "$curr_p3_dsk"
+
+                     #Clear GPT on new device, and create P3 partition
+                     clear_gpt "$1"
+                     create_new_p3_part "$1"
+
+                     #check the result
+                     new_p3_dev=$(/sbin/findfs PARTLABEL=P3)
+                     new_p3_dsk="/dev/"$(lsblk -no pkname "$new_p3_dev")
+                     [ "$new_p3_dsk" != "$1" ] && echo "Failed to attach persist to $1" && exit 1
```

Comment:
If this fails, shouldn't we relabled back? Or better yet -- should we relable for the first time only at this point?